app/views/cakes/_cake.html.haml

```diff
@@ -0,0 +1,6 @@
+%tr
+  %td
+    = link_to cake.name, edit_cake_path(cake)
+    %button= link_to "Usuń", cake_path(cake), method: :delete, data: {confirm: "Are you sure?"}
```

Comment:
Use a link or a button - never use both for one element. 
I think here it's better to go with a link styled as a button as <button> seems to be meant only for forms as submit button.